ios-trivia-game/ViewControllers/ResultsViewController.swift

```diff
@@ -45,12 +45,15 @@ class ResultsViewController: UIViewController {
         // populating the gameRoom Object
         FirebaseClient.instance.getGameBy(roomId: roomId!, complete: { (snapshot) in
             if let data = snapshot.value as? NSDictionary {
-                self.gameRoom = GameRoom(dictionary: data[data.allKeys.first as! String] as! NSDictionary)
+                self.gameRoom = GameRoom(dictionary: data)
                 
                 self.resultsTitleLabel.text = "Results: Round \(self.gameRoom!.current_question!) of \(self.gameRoom!.max_num_of_questions)"
                 
-                if (self.gameRoom?.current_question == self.gameRoom?.max_num_of_questions) {
-                    self.nextQuestionButton.isHidden = true
+                let curQues = self.gameRoom?.current_question
+                let maxQuestions = self.gameRoom?.max_num_of_questions
+                if (curQues! >= maxQuestions!) {
+                    // needs to be completed
+                    //self.nextQuestionButton.isHidden = true
```

Comment:
you need to hide the next question button if you've reached max number of questions no?